scripts/create_spec_repo/Sources/SpecRepoBuilder/main.swift

```diff
@@ -0,0 +1,301 @@
+  var github_account: String = "FirebasePrivate"
+  @Option(help: "Github Repo Name.")
+  var sdk_repo_name: String = "SpecsTesting"
+  @Option(help: "Local Podspec Repo Name.")
+  var local_spec_repo_name: String
+  @Flag(help: "Raise error while circular dependency detected.")
+  var raise_circular_dep_error: Bool = false
+  func generateOrderOfInstallation(pods: [String], podSpecDict: SpecFiles,
+                                   parentDeps: inout Set<String>) {
+    if podSpecDict.isEmpty() {
+      return
+    }
+
+    for pod in pods {
+      if !podSpecDict.contains(pod) {
+        continue
+      }
+      let deps = getTargetedDeps(of: pod, from: podSpecDict)
+      if parentDeps.contains(pod) {
+        print("Circular dependency is detected in \(pod) and \(parentDeps)")
+        if raise_circular_dep_error {
+          Self
+            .exit(withError: SpecRepoBuilderError
+              .circularDependencies(pods: parentDeps))
+        }
+        continue
+      }
+      parentDeps.insert(pod)
+      generateOrderOfInstallation(
+        pods: deps,
+        podSpecDict: podSpecDict,
+        parentDeps: &parentDeps
+      )
+      print("\(pod) depends on \(deps).")
+      podSpecDict.depInstallOrder.append(pod)
+      parentDeps.remove(pod)
+      podSpecDict.removeValue(forKey: pod)
+    }
+  }
+
+  func searchDeps(of pod: String, from podSpecFilesObj: SpecFiles) -> [String] {
+    var deps: [String] = []
+    var fileContents = ""
+    guard let podSpecURL = podSpecFilesObj.get(pod) else {
+      return deps
+    }
+    do {
+      fileContents = try String(contentsOfFile: podSpecURL.path, encoding: .utf8)
+    } catch {
+      fatalError("Could not read \(pod) podspec from \(podSpecURL.path).")
+    }
+    for line in fileContents.components(separatedBy: .newlines) {
+      if line.contains(_DEPENDENCY_LABEL_IN_SPEC) {
+        if _SKIP_LINES_WITH_WORDS.contains(where: line.contains) {
+          continue
+        }
+        let newLine = line.trimmingCharacters(in: .whitespacesAndNewlines)
+        let tokens = newLine.components(separatedBy: _DEPENDENCY_LINE_SEPARATORS)
+        if let depPrefix = tokens.first {
+          if depPrefix.hasSuffix(_DEPENDENCY_LABEL_IN_SPEC) {
+            let podNameRaw = String(tokens[1]).replacingOccurrences(of: "'", with: "")
+            if podNameRaw != pod { deps.append(podNameRaw) }
+          }
+        }
+      }
+    }
+    return deps
+  }
+
+  func filterTargetDeps(_ deps: [String], with targets: SpecFiles) -> [String] {
+    var targetedDeps: [String] = []
+    for dep in deps {
+      if targets.contains(dep) {
+        targetedDeps.append(dep)
+      }
+    }
+    return targetedDeps
+  }
+
+  func getTargetedDeps(of pod: String, from podSpecDict: SpecFiles) -> [String] {
+    let deps = searchDeps(of: pod, from: podSpecDict)
+    return filterTargetDeps(deps, with: podSpecDict)
+  }
+
+  func push_podspec(_ pod: String, from sdk_repo: String, sources: [String],
```

Comment:
Nit: same naming comments as before re: style and ambiguous arguments. A better name would likely be:

```
func pushPodspec(forPod podName: String,
                 sdkRepo: String,
                 sources: [String],
                 flags: [String],
                 shell: Shell = Shell.shared) -> Int32
```

Although the `sdkRepo` is a filepath, so a `URL` could be applicable as well.